src/Approximations/overapproximate.jl

```diff
@@ -516,84 +516,103 @@ function overapproximate(cap::Intersection{N,
 end
 
 """
-    overapproximate(lm::LinearMap{N, CartesianProductArray{N, T}},
-                      ::Type{CartesianProductArray{N, O}}) where {N, T<:LazySet{N}, O<:LazySet{N}}
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
+                    ::Type{CartesianProductArray{N, S}}
+                   ) where {N, S<:LazySet{N}}
 
-Decompose a lazy linear map of a cartesian product array keeping original block structure.
+Decompose a lazy linear map of a cartesian product array while keeping the
+original block structure.
 
 ### Input
 
-- `lm`   -- lazy linear map of cartesian product array
+- `lm` -- lazy linear map of cartesian product array
 - `CartesianProductArray` -- type for dispatch
 
 ### Output
 
 A `CartesianProductArray` representing the decomposed linear map.
 """
-function overapproximate(lm::LinearMap{N, CartesianProductArray{N, T}},
-                           ::Type{CartesianProductArray{N, O}}) where {N, T<:LazySet{N}, O<:LazySet{N}}
-
-    M, cpa = lm.M, lm.X
-    array = Vector{O}(undef, length(cpa.array))
-    return _overapproximate_lm_cpa!(M, cpa, array, O)
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
+                         ::Type{CartesianProductArray{N, S}}
+                        ) where {N, S<:LazySet{N}}
+    cpa = array(lm.X)
+    arr = Vector{S}(undef, length(cpa))
+    return _overapproximate_lm_cpa!(arr, lm.M, cpa, S)
 end
 
 """
-    overapproximate(lm::LinearMap{N, CartesianProductArray{N, T}},
-                      ::Type{CartesianProductArray{N, HPolytope{N}}},
-                   dir::AbstractDirections{N}) where {N, T<:LazySet{N}}
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
+                    ::Type{<:CartesianProductArray},
+                    dir::Type{<:AbstractDirections}) where {N}
 
-Decompose a lazy linear map of a cartesian product array.
+Decompose a lazy linear map of a cartesian product array with template
+directions while keeping the original block structure.
 
 ### Input
 
-- `lm`   -- lazy linear map of cartesian product array
+- `lm`  -- lazy linear map of a cartesian product array
 - `CartesianProductArray` -- type for dispatch
-- `dir` -- approximation option for decomposition
+- `dir` -- template directions for overapproximation
 
 ### Output
 
 A `CartesianProductArray` representing the decomposed linear map.
 """
-function overapproximate(lm::LinearMap{N, CartesianProductArray{N, T}},
-                           ::Type{<:CartesianProductArray},
-                         dir::Type{<:AbstractDirections}) where {N, T<:LazySet{N}}
-    M, cpa = lm.M, lm.X
-    array = Vector{HPolytope{N}}(undef, length(cpa.array))
-    return _overapproximate_lm_cpa!(M, cpa, array, dir)
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N}},
```

Comment:
This change is such that if the second argument is, for example `CartesianProductarray{N, HPolygon{N}}`, the `HPolygon` is ignored. Can you add this to the docstring?

Reply:
No, there is no functional change here. You could not specify the output type before either. The `T` parameter was just not used, so I removed it. We *always* allocate a `Vector{HPolytope{N}}` because template-direction overapproximation returns a `HPolytope{N}`.

Reply:
You're right! It is ignored in both cases. I opened https://github.com/JuliaReach/LazySets.jl/issues/1430 as follow up.